back/src/forms/resolvers/queries/formsLifeCycle.ts

```diff
@@ -160,17 +86,8 @@ const formsLifeCycleResolver: QueryResolvers["formsLifeCycle"] = async (
   const selectedCompany =
     userCompanies.find(uc => uc.siret === siret) || userCompanies.shift();
 
-  const formsFilter = {
-    OR: [
-      { owner: { id: user.id } },
-      { recipientCompanySiret: selectedCompany.siret },
-      { emitterCompanySiret: selectedCompany.siret },
-      {
-        transporterCompanySiret: selectedCompany.siret,
-        status: "SEALED"
-      }
-    ]
-  };
+  const formsFilter = getRolesFilter(selectedCompany.siret, []);
```

Comment:
Il y a peut-être une petite amélioration à faire au passage sur la signature de `getRolesFilter`. En effet je trouve bizarre de passer un tableau de rôles vide lorsque l'on demande tous les rôles comme c'est le cas ici. Je vois deux possibilités:
- définir une constante `const ALL_ROLES: FormRole[]  = [RECIPIENT, EMITTER, TRANSPORTER, TRADER, ECO_ORGANISME]` qui permettrait de faire `getRolesFilter(siret, ALL_ROLES)`
- rendre le second paramètre optionnel pour pouvoir faire `getRolesFilter(siret)` mais dans ce cas le nom de la fonction ne fait plus trop de sens et on pourrait changer ça en `getFormsFilter` 

```
/**
 * Filter forms in which the company is part of
 * Optional parameter roles allows to filter on specific roles
 * For example getFormsFilter(company, [TRANSPORTER]) will return a filter
 * only for the forms in which the company appears as a transporter
 */
export function getFormsFilter(
  siret: string,
  roles?: FormRole
): FormWhereInput {

  const filtersByRole = {
    ["RECIPIENT"]: [
      { recipientCompanySiret: siret },
      {
        temporaryStorageDetail: {
          destinationCompanySiret: siret
        }
      }
    ],
    ["EMITTER"]: [{ emitterCompanySiret: siret }],
    ["TRANSPORTER"]: [
      { transporterCompanySiret: siret },
      {
        transportSegments_some: {
          transporterCompanySiret: siret
        }
      },
      {
        temporaryStorageDetail: {
          transporterCompanySiret: siret
        }
      }
    ],
    ["TRADER"]: [{ traderCompanySiret: siret }],
    ["ECO_ORGANISME"]: [{ ecoOrganisme: { siret: siret } }]
  };

  if (roles) {
    return {
      OR: Object.keys(filtersByRole)
        .filter(role => roles.includes(role))
        .map(role => filtersByRole[role])
        .flat()
    };
  }

  // all roles
  return { OR: Object.values(filtersByRole).flat() };
}
```

Par ailleurs je pense que ce serait chouette pour l'organisation du code de ne pas importer dans un resolver de fonctions définies dans une autre resolver. Si besoin il y a de mutualiser une fonction comme c'est le cas ici, on peut la déplacer dans un fichier commun comme `database.ts`.

Reply:
Why not, par contre du coup c'est un breaking change. Si certains envoient un tableau vide actuellement ils ne vont plus avoir de résultat.

Reply:
Remarque la doc dans le chéma dit:
> Si aucune filtre n'est passé, les bordereaux seront retournés quel que soit votre role dessus.
    Défaut à vide.

Ca ne sous entend donc pas qu'un tableau vide récupère tout